crates/core/src/credits.rs

```diff
@@ -166,6 +166,7 @@ impl<I: LineItem> CreditsClient<I> {
         self.core
             .credit_sheet
             .get(key)
+            .and_then(Option::as_ref)
             .ok_or_else(|| anyhow!("No price in credit sheet found for key {key:?}"))
```

Comment:
Can we get unique error enum for the CreditClient like to have separate errors to match on for no entry and another for not enough credits.

Or maybe allow None for when the credit line item doesn't exist and return error when user doesnt have enough credit. That seems to fit better